processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -461,15 +461,25 @@ func (c *WatchClient) getPodFromAPI(pod *api_v1.Pod) *Pod {
 }
 
 func (c *WatchClient) getIdentifiersFromAssoc(newPod *Pod) []PodIdentifier {
+	if newPod.Attributes == nil {
+		newPod.Attributes = map[string]string{}
+	}
+
 	ids := []PodIdentifier{}
 	for _, assoc := range c.Associations {
-		ret := []string{}
-		for _, source := range assoc.Sources {
+		ret := PodIdentifier{}
+		skip := false
+		for i, source := range assoc.Sources {
 			// If association configured to take IP address from connection
 			switch {
 			case source.From == "connection":
-				if newPod.Address != "" {
-					ret = append(ret, newPod.Address)
+				if newPod.Address == "" {
+					skip = true
+				} else {
+					ret[i] = GetPodIdentifierAttribute(source, newPod.Address)
+					if source.Name != "" {
+						newPod.Attributes[source.Name] = newPod.Address
+					}
```

Comment:
Why is this needed? I think `source.Name` should be ignored if `source.From == "connection"`

Reply:
Hmm, maybe it wasn't `source.Name` being used, just always `k8s.pod.ip` :sweat_smile: